server/plugin/taskLinkPreview.go

```diff
@@ -106,3 +106,52 @@ func (p *Plugin) PostPullRequestPreview(linkData []string, link, userID, channel
 
 	return post, ""
 }
+
+func (p *Plugin) PostBuildDetailsPreview(linkData []string, link, userID, channelID string) (*model.Post, string) {
+	organization := linkData[3]
+	project := linkData[4]
+	buildID := strings.Split(linkData[6], "&")[0][16:]
+	buildDetails, _, err := p.Client.GetBuildDetails(organization, project, buildID, userID)
+	if err != nil {
+		return nil, ""
+	}
+
+	post := &model.Post{
+		UserId:    userID,
+		ChannelId: channelID,
+	}
+
+	attachment := &model.SlackAttachment{
+		AuthorName: "Azure Pipeline",
+		AuthorIcon: fmt.Sprintf("%s/plugins/%s/static/%s", p.GetSiteURL(), constants.PluginID, constants.FileNamePipeline), // TODO: update icon file
```

Comment:
You should put the icon in the `public` folder and use this URL - `<siteurl>/plugins/<pluginID>/public/<filename>`

Reply:
we have kept bot images inside the `asset` folder and are hosting that folder as well so I think keeping them here also does the same thing, pls let me know if we have other reasons as well to use `public`

Reply:
Keeping files and images in `assets` folder requires us to host the files explicitly by writing some code which must be written in your `api.go` file. But we can remove that code and make use of the code that MM has already written by keeping files in the public folder.

Reply:
we have logged this ticket to do this separately once everything gets merged to avoid changing this across different branches/conflicts
https://brightscout.atlassian.net/browse/MI-2462